src/s2e/Plugins/Lua/LuaInstructionAnnotation.cpp

```diff
@@ -177,6 +179,29 @@ void LuaInstructionAnnotation::onInstruction(S2EExecutionState *state, uint64_t
     if (luaAnnotation.exitCpuLoop()) {
         throw CpuExitException();
     }
+
+    if (luaAnnotation.doSkip()) {
```

Comment:
I would rename the api to skipInstruction (instead of setSkip). It's more user friendly.